app/backend/core/authentication.py

```diff
@@ -237,3 +257,74 @@ async def check_path_auth(self, path: str, auth_claims: dict[str, Any], search_c
             break
 
         return allowed
+
+    # See https://github.com/Azure-Samples/ms-identity-python-on-behalf-of/blob/939be02b11f1604814532fdacc2c2eccd198b755/FlaskAPI/helpers/authorization.py#L44
+    async def validate_access_token(self, token: str):
+        """
+        Validate an access token is issued by Entra
+        """
+        jwks = None
+        async for attempt in AsyncRetrying(
```

Comment:
It looks like you added the tenacity retry, is that because you found it flaky in your experience?

Reply:
The sample had retries using session adapters, so I used the existing retry pattern in our repo instead